examples/apps/generic_client/direct_model_generic_client.py

```diff
@@ -0,0 +1,265 @@
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import json
+import logging
+import random
+
+from avalon_sdk.worker.worker_details import WorkerType, WorkerStatus
+import avalon_sdk.worker.worker_details as worker_details
+from avalon_sdk.connector.blockchains.ethereum.ethereum_worker_registry_list \
+    import EthereumWorkerRegistryListImpl
+from avalon_sdk.connector.direct.jrpc.jrpc_worker_registry import \
+    JRPCWorkerRegistryImpl
+from avalon_sdk.connector.direct.jrpc.jrpc_work_order import \
+    JRPCWorkOrderImpl
+from avalon_sdk.connector.direct.jrpc.jrpc_work_order_receipt \
+    import JRPCWorkOrderReceiptImpl
+from error_code.error_status import WorkOrderStatus, ReceiptCreateStatus
+import avalon_crypto_utils.signature as signature
+from error_code.error_status import SignatureStatus
+from avalon_sdk.work_order_receipt.work_order_receipt \
+    import WorkOrderReceiptRequest
+from base_generic_client import BaseGenericClient
+
+logging.basicConfig(
+    format="%(asctime)s - %(levelname)s - %(message)s", level=logging.INFO)
+
+
+class DirectModelGenericClient(BaseGenericClient):
+    """
+    Generic client class to test end to end flow
+    for direct model.
+    """
+
+    def __init__(self, config):
+        super().__init__()
+        self._config = config
+        self._worker_registry_list = None
+        self._worker_instance = JRPCWorkerRegistryImpl(self._config)
+        self._work_order_instance = JRPCWorkOrderImpl(self._config)
+        self._work_order_receipt = JRPCWorkOrderReceiptImpl(self._config)
+
+    def _get_random_jrpc_id(self):
+        return random.randint(1, 10000)
+
+    def retrieve_uri_from_registry_list(config):
+        # Retrieve Http JSON RPC listener uri from registry
+        logging.info("\n Retrieve Http JSON RPC listener uri from registry \n")
+        # Get block chain type
+        blockchain_type = config['blockchain']['type']
+        if blockchain_type == "Ethereum":
+            self._worker_registry_list = EthereumWorkerRegistryListImpl(
+                config)
+        else:
```

Comment:
Do we support registry uri in case of Fabric ?

Reply:
No